### PR TITLE
scbuild: fix unhandled case during DROP SCHEMA

### LINKED ISSUE #133510: sql/tests: TestRandomSyntaxGeneration failed [unhandled kind: 0]
sql/tests.TestRandomSyntaxGeneration [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/17460885?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/17460885?buildTab=artifacts#/) on release-23.2 @ [520b62634e9478db36c5ec81d43d1ccaf821185c](https://github.com/cockroachdb/cockroach/commits/520b62634e9478db36c5ec81d43d1ccaf821185c):

Random syntax error:

```
    rsg_test.go:962: Crash detected: server panic: pq: internal error: building declarative schema change targets for DROP SCHEMA: unhandled kind: 0
```
Query:

```
        DROP SCHEMA PUBLIC;
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #132558 sql/tests: TestRandomSyntaxGeneration failed [C-test-failure O-robot T-db-server branch-release-24.1.6-rc]
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRandomSyntaxGeneration.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

Jira issue: CRDB-43649

### LINKED ISSUE #133913: sql/tests: TestRandomSyntaxGeneration failed [unhandled kind: 0]
sql/tests.TestRandomSyntaxGeneration [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/17523456?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/17523456?buildTab=artifacts#/) on master @ [becbd0fcdfa2e37a6ff23b33af70f2f91eca0790](https://github.com/cockroachdb/cockroach/commits/becbd0fcdfa2e37a6ff23b33af70f2f91eca0790):

Random syntax error:

```
    rsg_test.go:962: Crash detected: server panic: pq: internal error: building declarative schema change targets for DROP SCHEMA: unhandled kind: 0
```
Query:

```
        DROP SCHEMA IF EXISTS ident , ident . PUBLIC , WORK , POLYGON . ident RESTRICT;
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #133510 sql/tests: TestRandomSyntaxGeneration failed [C-test-failure O-robot P-2 T-sql-foundations branch-release-23.2]
- #132558 sql/tests: TestRandomSyntaxGeneration failed [C-test-failure O-robot P-2 T-db-server branch-release-24.1.6-rc]
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRandomSyntaxGeneration.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

Jira issue: CRDB-43820

### LINKED ISSUE #134280: sql/tests: TestRandomSyntaxGeneration failed [building declarative schema change targets for DROP SCHEMA: unhandled kind: 0]
sql/tests.TestRandomSyntaxGeneration [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/17596372?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/17596372?buildTab=artifacts#/) on release-24.3 @ [25385fc6471c86de3f145c5990e7e9c0f28fbb9a](https://github.com/cockroachdb/cockroach/commits/25385fc6471c86de3f145c5990e7e9c0f28fbb9a):

Random syntax error:

```
    rsg_test.go:962: Crash detected: server panic: pq: internal error: building declarative schema change targets for DROP SCHEMA: unhandled kind: 0
```
Query:

```
        DROP SCHEMA IF EXISTS PUBLIC , SEARCH;
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #133913 sql/tests: TestRandomSyntaxGeneration failed [unhandled kind: 0] [C-test-failure O-robot T-sql-foundations branch-master]
- #133510 sql/tests: TestRandomSyntaxGeneration failed [unhandled kind: 0] [C-test-failure O-robot P-2 T-sql-foundations branch-release-23.2]
- #132558 sql/tests: TestRandomSyntaxGeneration failed [C-test-failure O-robot P-2 T-db-server branch-release-24.1.6-rc]
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRandomSyntaxGeneration.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

Jira issue: CRDB-43992

### LINKED ISSUE #134752: sql/tests: TestRandomSyntaxGeneration failed [building declarative schema change targets for DROP SCHEMA: unhandled kind: 0]
sql/tests.TestRandomSyntaxGeneration [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/17671416?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/17671416?buildTab=artifacts#/) on release-24.2 @ [5764366875adca70c3a3ff3d4d13aaf599bc53a7](https://github.com/cockroachdb/cockroach/commits/5764366875adca70c3a3ff3d4d13aaf599bc53a7):

Random syntax error:

```
    rsg_test.go:962: Crash detected: server panic: pq: internal error: building declarative schema change targets for DROP SCHEMA: unhandled kind: 0
```
Query:

```
        DROP SCHEMA IF EXISTS REPLACE , IMMUTABLE , ident . PUBLIC RESTRICT;
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #134280 sql/tests: TestRandomSyntaxGeneration failed [building declarative schema change targets for DROP SCHEMA: unhandled kind: 0] [C-test-failure O-robot P-2 T-sql-foundations branch-release-24.3]
- #133913 sql/tests: TestRandomSyntaxGeneration failed [unhandled kind: 0] [C-test-failure O-robot P-2 T-sql-foundations branch-master]
- #133510 sql/tests: TestRandomSyntaxGeneration failed [unhandled kind: 0] [C-test-failure O-robot P-2 T-sql-foundations branch-release-23.2]
- #132558 sql/tests: TestRandomSyntaxGeneration failed [CREATE TENANT statement is not valid after parsing then formatting] [C-test-failure O-robot P-2 T-db-server branch-release-24.1.6-rc]
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRandomSyntaxGeneration.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

Jira issue: CRDB-44213

### LINKED ISSUE #135141: sql/tests: TestRandomSyntaxGeneration failed [unhandled kind: 0]
sql/tests.TestRandomSyntaxGeneration [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/17753154?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/17753154?buildTab=artifacts#/) on release-24.1 @ [6a1ab2b91f02d6f1680b988df8d2d3828b2cfa28](https://github.com/cockroachdb/cockroach/commits/6a1ab2b91f02d6f1680b988df8d2d3828b2cfa28):

Random syntax error:

```
    rsg_test.go:962: Crash detected: server panic: pq: internal error: building declarative schema change targets for DROP SCHEMA: unhandled kind: 0
```
Query:

```
        DROP SCHEMA PUBLIC;
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #134752 sql/tests: TestRandomSyntaxGeneration failed [building declarative schema change targets for DROP SCHEMA: unhandled kind: 0] [C-test-failure O-robot P-2 T-sql-foundations branch-release-24.2]
- #134280 sql/tests: TestRandomSyntaxGeneration failed [building declarative schema change targets for DROP SCHEMA: unhandled kind: 0] [C-test-failure O-robot P-2 T-sql-foundations branch-release-24.3]
- #133913 sql/tests: TestRandomSyntaxGeneration failed [unhandled kind: 0] [C-test-failure O-robot P-2 T-sql-foundations branch-master]
- #133510 sql/tests: TestRandomSyntaxGeneration failed [unhandled kind: 0] [C-test-failure O-robot P-2 T-sql-foundations branch-release-23.2]
- #132558 sql/tests: TestRandomSyntaxGeneration failed [CREATE TENANT statement is not valid after parsing then formatting] [C-test-failure O-robot P-2 T-db-server branch-release-24.1.6-rc]
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRandomSyntaxGeneration.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

Jira issue: CRDB-44370

### PR DESCRIPTION
**Pull request by rafiss**
fixes https://github.com/cockroachdb/cockroach/issues/135141
fixes https://github.com/cockroachdb/cockroach/issues/133913
fixes https://github.com/cockroachdb/cockroach/issues/134752
fixes https://github.com/cockroachdb/cockroach/issues/134280
fixes https://github.com/cockroachdb/cockroach/issues/133510

Release note (bug fix): Fixed an unhandled error that would occur if DROP SCHEMA was executed on the `public` schema of the `system` database, or on an internal schema like `pg_catalog` or `information_schema`.

### CODE REVIEW

**Review by annrpom — APPROVED** [review]
nice! lgtm; just had a small q

File: pkg/sql/drop_schema.go [hunk 1]
```diff
@@ -70,6 +70,14 @@ func (p *planner) DropSchema(ctx context.Context, n *tree.DropSchema) (planNode,
 			}
 			return nil, pgerror.Newf(pgcode.InvalidSchemaName, "unknown schema %q", scName)
 		}
+		hasOwnership, err := p.HasOwnership(ctx, sc)
+		if err != nil {
+			return nil, err
+		}
+		if !hasOwnership {
+			return nil, pgerror.Newf(pgcode.InsufficientPrivilege,
+				"must be owner of schema %s", tree.Name(sc.GetName()))
```
[the remark this answers is not in the document]
Comment (annrpom) [reply pkg/sql/drop_schema.go:79]: or any of the internal schema like `pg_catalog`

[the remark this answers is not in the document]
Comment (rafiss) [reply pkg/sql/drop_schema.go:79]: yeah, the `node` user is the owner. it's a reserved role name that is used for executing internal operations and for owning objects that are baked into the database; and it's not possible to login as `node`.

**Comment by rafiss** [comment]
tftr!

bors r+

**Pull request merged**